### PR TITLE
make sure to export audio apis in exported applab apps

### PR DESCRIPTION
**Pull request by pcardune**

### CODE REVIEW

**Comment by pcardune** [comment]
@islemaster looks like this was caused by the changes in #12543. I'm thinking there must be a better way to get the complete list of all commands exposed as global functions, but I don't understand how all the different command configs get hooked up. Do you have any suggestions on that front? While the bug is fixed in this PR, I think the fix is rather brittle.

**Comment by islemaster** [comment]
I think there are already a couple of other things that might be missing here - the [dontMarshalApi](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/applab/dontMarshalApi.js) which also gets referenced in applab/dropletConfig.js, and the [makerlab api.js](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/makerlab/api.js) which gets referenced from [makerlab's dropletconfig](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/makerlab/dropletConfig.js) which gets [mixed in to applab's dropletconfig in some cases](https://github.com/code-dot-org/code-dot-org/blob/4a05a9526fcd5b8749691cb90ed2d3fd48e168aa/apps/src/applab/applab.js#L683-L687).

Maybe it would be more reliable to run through the final dropletConfig and look at the unique `parent`s of the various passed commands?  I also think this whole droplet setup is in need of some refactoring anyway - I'd like to see more subsets of commands broken out into small modules like makerlab and the audio commands, but we don't really have a standard approach to that yet.

**Comment by pcardune** [comment]
Ok, I refactored things a bit and added tests for all the functions exposed by dropletConfig. The makerlab apis are still unsupported at this point. I haven't played with that stuff at all yet.

**Comment by islemaster** [comment]
What's with all the updated localizations in the diffs?

**Comment by pcardune** [comment]
accidentally pulled from staging even though this PR was not against staging. Switched the base of this PR to staging and now it looks right.

File: apps/test/unit/applab/exportTest.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        //   var str = "Hello World";
+        return;
+      }
+      it(`exposes a ${block.func} function`, () => {
+        if (globals.indexOf(block.func) < 0) {
+          if (!eval(`window.${block.func}`)) { //eslint-disable-line no-eval
+            assert(false, `expected exported globals to include ${block.func}`);
+          }
+        }
+      });
```
Comment (islemaster) [inline apps/test/unit/applab/exportTest.js:26]: This is pretty sweet - requiring internal consistency between the `dropletConfig` and the exported globals.  But there's something I don't understand about the export process here - do these necessarily need to be global?  In `dropletConfig` most entries have a `parent` object where they'll supposedly find the method they're looking for, without it being on `window` AFAIK.

File: apps/src/applab/export.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import * as api from './api';
+import {executors} from '../lib/util/audioApi';
+import * as dontMarshalApi from './dontMarshalApi';
+
+export function getExportedGlobals() {
+  return Object.assign({}, executors, dontMarshalApi, api);
```
Comment (islemaster) [inline apps/src/applab/export.js:6]: :+1: To isolating this weirdness into its own file.  It'd be nice to have a comment explaining what this is for though - it's not obvious to me in isolation.

File: apps/test/unit/applab/exportTest.js (hunk 1 above)
Comment (pcardune) [reply apps/test/unit/applab/exportTest.js:26]: exported apps do not run through our interpreter. Instead, you just end up with a script tag pointing to a file with the student's js code. As a result, and functions that the student's code might call to need to be on `window`.

Comment (islemaster) [reply apps/test/unit/applab/exportTest.js:26]: Got it.  _Great_ test then!

**Comment by islemaster** [comment]
LGTM!

**Pull request merged**